server/src/main/java/io/unitycatalog/server/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,119 @@
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import lombok.SneakyThrows;
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Security settings for the Authnz framework.
+ *
+ * <p>The settings are loaded from files persisted in the UC configuration folder. If no settings
+ * files exist, they will be generated for reuse across restarts. Equivalent openssl commands are -
+ * openssl genrsa -out private_key.pem 2048 - openssl rsa -in private_key.pem -pubout -outform DER
+ * -out public_key.der - openssl pkcs8 -topk8 -inform PEM -outform DER -in private_key.pem -out
+ * private_key.der -nocrypt - openssl rand -hex -out key_id.txt 32
+ */
+public class SecurityConfiguration {
+
+  private static final Logger log = LoggerFactory.getLogger(SecurityConfiguration.class);
+
+  private Path rsa512PublicKey;
+
+  private Path rsa512PrivateKey;
+
+  private Path keyId;
+
+  private String rsa512PublicKeyEncoded = null;
+
+  private String rsa512PrivateKeyEncoded = null;
+
+  @SneakyThrows
+  public SecurityConfiguration(Path configurationFolder) {
+    rsa512PublicKey = configurationFolder.resolve("public_key.der");
+    rsa512PrivateKey = configurationFolder.resolve("private_key.der");
+    keyId = configurationFolder.resolve("key_id.txt");
+
+    initializeIfMissing();
+  }
+
+  @SneakyThrows
+  public void initializeIfMissing() {
+    if (Files.notExists(rsa512PublicKey)
+        || Files.notExists(rsa512PrivateKey)
+        || Files.notExists(keyId)) {
+      log.info("Initializing security configuration.");
+      KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+      keyPairGenerator.initialize(2048);
+      KeyPair keyPair = keyPairGenerator.generateKeyPair();
+
+      Files.write(rsa512PublicKey, keyPair.getPublic().getEncoded(), StandardOpenOption.CREATE);
+      Files.write(rsa512PrivateKey, keyPair.getPrivate().getEncoded(), StandardOpenOption.CREATE);
+
+      byte[] keyIdBytes = new byte[32];
+      new SecureRandom().nextBytes(keyIdBytes);
+      Files.writeString(keyId, Hex.encodeHexString(keyIdBytes), StandardOpenOption.CREATE);
+    }
+  }
+
+  public Algorithm algorithmRSA()
+      throws NoSuchAlgorithmException, InvalidKeySpecException, IOException {
+    return Algorithm.RSA512(rsaPublicKey(), rsaPrivateKey());
+  }
+
+  public RSAPublicKey rsaPublicKey()
+      throws NoSuchAlgorithmException, InvalidKeySpecException, IOException {
+    byte[] keyBytes;
+    if (rsa512PublicKeyEncoded != null) {
+      keyBytes = Base64.getDecoder().decode(rsa512PublicKeyEncoded);
+    } else {
+      if (Files.notExists(rsa512PublicKey)) {
+        log.info("No JWT public signing key present.");
+        return null;
+      }
+      keyBytes = Files.readAllBytes(rsa512PublicKey);
```

Comment:
nit: couldn't lines 83-90 be simplified by moving this read (and the corresponding logic in `rsaPrivateKey`) to line 54? 

Line 53 is there to handle the case of missing key files and the invariant at line 54 is that the key files exist, otherwise the service can panic.